modules/tinymce/src/core/main/ts/Rtc.ts

```diff
@@ -0,0 +1,296 @@
+import { Event, Node as DomNode, Range } from '@ephox/dom-globals';
+import { Fun, Obj, Option, Type } from '@ephox/katamari';
+import Editor from './api/Editor';
+import Node from './api/html/Node';
+import Serializer from './api/html/Serializer';
+import * as FilterNode from './html/FilterNode';
+import * as Operations from './undo/Operations';
+import { Index, Locks, UndoBookmark, UndoLevel, UndoLevelType, UndoManager } from './undo/UndoManagerTypes';
+import * as ApplyFormat from './fmt/ApplyFormat';
+import * as RemoveFormat from './fmt/RemoveFormat';
+import * as ToggleFormat from './fmt/ToggleFormat';
+import { RangeLikeObject } from './selection/RangeTypes';
+import { FormatRegistry } from './fmt/FormatRegistry';
+import { Content } from './content/EditorContent';
+import { GetContentArgs, ContentFormat, getContentInternal } from './content/GetContentImpl';
+import { SetContentArgs, setContentInternal } from './content/SetContentImpl';
+import { insertHtmlAtCaret } from './content/InsertContentImpl';
+import { getSelectedContentInternal } from './selection/GetSelectionContentImpl';
+
+const isTreeNode = (content: any): content is Node => content instanceof Node;
+
+const isSupportedContentFormat = (format: string) => format !== 'text';
```

Comment:
This is a bit confusing, because as far as I can see it's just used for RTC? Maybe it should be called `isSupportedRtcContentFormat`? I also have to wonder if this should be defined in the adaptor implementation, rather than hard coded into core?

Additionally it allows all formats, which seems wrong, as does that mean it works for bbcode, tree, raw and HTML? As such I'd expect this to be a whitelist, rather than a blacklist.

Reply:
It's not exported so we can call it whatever we want, really.

I think the idea was that it's a blacklist because RTC might add new format types e.g. `json`, but you're right, if that is the intent it needs to be more extensive.

Having two calls into the plugin is fine, so thinking about your feedback
* a blacklist stays here, in case TinyMCE adds more formats to blacklist
* a whitelist can move to the RTC plugin, so if it adds new capabilities it's able to override

I don't really mind which way we go, @spocke any thoughts?

Reply:
If guess it could check for undefined and html and fallback on text and fail on anything else. But then we would have to update this for json later not sure we need to support tree or raw they are not supposed to be used much.

Reply:
This is sounding more and more like this logic should be handled via the `RtcRuntimeApi`, as it's really up to the RTC implementation as to what it supports.

Reply:
Yes let's do that we can get the text content in rtc instead pulling it from view makes nice new lines for blocks though but we could do that in the rtc core.

Reply:
I was thinking we just move the "is supported" check to RTC so we can continue to fall back to text support in core as we do now? or does that not make sense? 🤔

Reply:
Removed the function and made explicit ifs instead easier to read and fix those corner cases.